### PR TITLE
[16/N] Fix clang-tidy warnings in jit

### PR DESCRIPTION
**Pull request by cyyever**
Follows #132564

cc @EikanWang @jgong5

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -1872,10 +1875,10 @@ TensorExprKernel::TensorExprKernel(
       symbolic_shape_inputs_(std::move(symbolic_shape_inputs)),
       custom_lowerings_(std::move(custom_lowerings)),
       pre_alloc_(pre_alloc),
-      kernel_func_name_(kernel_func_name),
+      kernel_func_name_(std::move(kernel_func_name)),
       symbolic_strides_(std::move(symbolic_strides)) {
   optimizeOwningGraph();
-
+  // NOLINTNEXTLINE
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/kernel.cpp:1881]: What's the lint failure on the next line?

File: torch/csrc/jit/tensorexpr/kernel.h [hunk 2]
```diff
@@ -9,9 +9,9 @@
 #include <torch/csrc/jit/tensorexpr/lowerings.h>
 #include <torch/csrc/jit/tensorexpr/tensor.h>
 
-namespace torch {
-namespace jit {
-namespace tensorexpr {
+#include <utility>
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/kernel.h:12]: One again, put includes at top not in the nested scope.

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp [hunk 3]
```diff
@@ -451,7 +454,7 @@ void initTensorExprBindings(PyObject* module) {
   py::class_<LoopNest>(te, "LoopNest")
       .def(py::init<const std::vector<Tensor>&>())
       .def(py::init<const std::vector<Tensor>&, const std::vector<Tensor>&>())
-      .def(py::init([](StmtPtr s, const std::vector<BufHandle>& bufs) {
+      .def(py::init([](const StmtPtr& s, const std::vector<BufHandle>& bufs) {
         std::unordered_set<BufPtr> buf_nodes;
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:458]: Can you add a reserve here while you are editing this?

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp [hunk 4]
```diff
@@ -508,77 +513,83 @@ void initTensorExprBindings(PyObject* module) {
...
+          [](const ForPtr& f, int factor) {
             ForPtr inner = nullptr;
             LoopNest::splitWithMask(f, factor, &inner);
             return inner;
           },
           py::return_value_policy::reference)
       .def(
           "slice_head",
-          [](ForPtr f, int factor) {
+          [](const ForPtr& f, int factor) {
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:549]: Nit some of the makes_tuples should have their args moved into position for return.

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Lots of nits and changes with this one

File: torch/csrc/jit/tensorexpr/kernel.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/kernel.cpp:1881]: Fixed

File: torch/csrc/jit/tensorexpr/kernel.h (hunk 2 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/kernel.h:12]: Fixed

File: torch/csrc/jit/tensorexpr/registerizer.h [hunk 5]
```diff
@@ -250,7 +248,7 @@ class Scope {
   const std::unordered_set<VarPtr>& localVars() const {
     return localVars_;
   }
-  void addLocalVar(VarPtr v) {
+  void addLocalVar(const VarPtr& v) {
     localVars_.insert(v);
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/registerizer.h:252]: Fixed

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp (hunk 3 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:458]: Fixed

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp [hunk 6]
```diff
@@ -752,26 +765,27 @@ void initTensorExprBindings(PyObject* module) {
...
-                g, custom_lowerings, symbolic_shape_inputs, pre_alloc);
-          }),
+          py::init(
+              [](const TSGraph& g,
+                 const std::unordered_map<std::string, NNCLoweringFunction>&
+                     custom_lowerings_str,
+                 const std::vector<int64_t>& symbolic_shape_inputs,
+                 bool pre_alloc = false) {
+                std::unordered_map<c10::Symbol, NNCLoweringFunction>
+                    custom_lowerings;
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:782]: Fixed

File: torch/csrc/jit/tensorexpr/tensorexpr_init.cpp [hunk 7]
```diff
@@ -752,26 +765,27 @@ void initTensorExprBindings(PyObject* module) {
...
+                for (auto& kv : custom_lowerings_str) {
+                  custom_lowerings[c10::Symbol::fromQualString(kv.first)] =
+                      kv.second;
+                }
+                return std::make_unique<TensorExprKernel>(
+                    g, custom_lowerings, symbolic_shape_inputs, pre_alloc);
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/tensorexpr_init.cpp]: Fixed

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 8]
```diff
@@ -1859,23 +1857,23 @@ void TensorExprKernel::recompile() {
...
     std::unordered_map<c10::Symbol, NNCLoweringFunction> custom_lowerings,
     std::vector<int64_t> symbolic_shape_inputs,
     bool pre_alloc /*= false*/,
     std::unordered_map<
         const torch::jit::Value*,
         std::vector<torch::jit::StrideInput>> symbolic_strides)
-    : graph_(subgraph),
-      code_(subgraph, ""),
+    : graph_(std::move(subgraph)),
+      code_(graph_, ""),
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/kernel.cpp:1869]: This is a bit dangerous as members are initialized in the order they are declared in the class, not the order they are listed in here so it could cause unexpected behavior if the class is refactored. I would revert the changes to subgraph to reduce maintenance burden

**Review by Skylion007 — CHANGES_REQUESTED** [review]
Fix the one major block, it's not that safe to rely on member initializer order (which is the order they appear in the C++ class). I thought their was a clang-tidy rule that flagged this, but maybe it's not enabled.

File: torch/csrc/jit/tensorexpr/kernel.cpp (hunk 8 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/kernel.cpp:1869]: Understand your concern, I'm fine with reverting.

**Review by Skylion007 — APPROVED** [review]
